docs/getting-started/6-proofs.md

```diff
@@ -1,3 +1,130 @@
+
+## Method 1 - Prover (holder) responds to presentation proposal from the verifier
+
+> Note: This setup is assumed for a react native mobile agent
+
+> Note: This process assumes there is an established connection between the prover and the verifier
+
+### 1. Configure agent
+
+Please make sure you reviewed the [agent setup overview](0-agent.md).
+
+### 2. Configure proof events handler
+
+This handler will be triggered whenever there is a Proof state change.
+
+```ts
+const handleProofStateChange = async (agent: Agent, event: ProofStateChangedEvent) => {
+  console.log(
+    `>> Proof state changed: ${event.payload.proofRecord.id}, previous state -> ${event.payload.previousState} new state: ${event.payload.proofRecord.state}`
+  )
+
+  if (event.payload.proofRecord.state === ProofState.RequestReceived) {
+    const retrievedCredentials = await agent.proofs.getRequestedCredentialsForProofRequest(
+      event.payload.proofRecord.id,
+      {
+        filterByPresentationPreview: true,
+      }
+    )
+
+    const requestedCredentials = agent.proofs.autoSelectCredentialsForProofRequest(retrievedCredentials)
+
+    agent.proofs.acceptRequest(event.payload.proofRecord.id, requestedCredentials)
+  }
+}
+```
+
+- `filterByPresentationPreview`: Whether to filter the retrieved credentials using the presentation preview. This configuration will only have effect if a presentation proposal message is available containing a presentation preview..
+
+Make sure to add the event listener to the agent after initializing the wallet
+
+```ts
+agent.events.on<ProofStateChangedEvent>(ProofEventTypes.ProofStateChanged, (event) => {
+  handleProofStateChange(event)
+})
+```
+
+To interfere the process and display an optional message to the user
+
+replace
+
+```ts
+agent.proofs.acceptRequest(event.payload.proofRecord.id, requestedCredentials)
+```
+
+by
+
+```ts
+Alert.alert('Attention!', message, [
+  {
+    text: 'Accept',
+    onPress: () => {
+      //Respond by
+      agent.proofs.acceptRequest(event.payload.proofRecord.id, requestedCredentials)
+    },
+  },
+  {
+    text: 'Reject',
+    onPress: () => {
+      console.log('User rejected offer')
+    },
+  },
+])
+```
+
+To loop through requested attributes and display some context to the user
+
+```ts
+const proofRequest = event.payload.proofRecord.requestMessage?.indyProofRequest
+var message = '>> Proof Request Recieved <<\n'
+message += `To prove:${proofRequest?.name}\n`
+message += 'Attributes to prove:\n'
+
+Object.values(proofRequest.requestedAttributes).forEach((attr) => {
+  message += `${attr.name}\n`
+})
+
+message += `Accept proof request?`
+```
+
+By sending the response to the verifier, the verifier will go through the process of verifying the VC and respond with an ack message.
+To give some context to the user you can add the following code to the Proof event handler
+
+```ts
```

Comment:
I feel like all the different code blocks make it quite hard to understand how they fit together. I think first making a full e2e flow in a single code block that is as simple as possible (so no `message = `, `console.log` etc...) and then add different sections going into more detail on specific topics such getting the requested values, auto selecting credentials, manually selecting credentials, ...

Reply:
sounds good

Reply:
done